### PR TITLE
IGNITE-15508 Added loading via ServiceLoader for JDBC Driver

### PR DESCRIPTION
**Pull request by vladErmakov07**
https://issues.apache.org/jira/browse/IGNITE-15508

### CODE REVIEW

**Comment by AMashenkov** [comment]
Please, fix IgniteJdbcDriver class javadoc.
as there will be no caches in Ignite 3.

File: modules/client/src/main/java/org/apache/ignite/jdbc/IgniteJdbcDriver.java [hunk 1]
```diff
@@ -107,7 +107,7 @@
      * @return Driver instance.
      * @throws RuntimeException when failed to register driver.
      */
-    public static synchronized Driver register() {
+    private static synchronized Driver register() {
```
Comment (AMashenkov) [inline modules/client/src/main/java/org/apache/ignite/jdbc/IgniteJdbcDriver.java:110]:
```suggestion
    private static synchronized void register() {
```

Comment (AMashenkov) [reply modules/client/src/main/java/org/apache/ignite/jdbc/IgniteJdbcDriver.java:110]: Let's drop 'registered' flag and make instance non-final.
See: https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/Driver.java

**Comment by vladErmakov07** [comment]
> Please, fix IgniteJdbcDriver class javadoc.
> as there will be no caches in Ignite 3.

I fixed it, thanks.

**Pull request merged**